Crayons: buttons followup (#15720)
forem/forem: app/javascript/crayons/Buttons/__stories__/Buttons.mdx

For example, a 'Submit' control on 'Sign in' view (form) would suit a `<Button />` because its job is to submit the form. 
However, 'Register' call-to-action—taking the user to a different form, in a different page or screen—would benefit from `<CTA />` or `<Link />` semantics.

Do not use `<Button />s` as navigational elements. Instead, use `<Link />s` (or `<CTA />s`) when the desired action is to take the user to a new page.

## Variants

### Default

> _Default_ variant is not very prominent, and therefore can be used for less prominent actions.

Reviewer: Wording (blocking): Incomplete sentence. Start with `The`. There are other instances of this below as well in the rest of the documentation.

```suggestion
The _Default_ variant is not prominent, and therefore can be used for less prominent actions.
```

Wording (nitpick):

```suggestion
The _Default_ variant is not prominent. Use it potentially for less noticeable actions.
```

Author: _The_ vs _A_ was always my weak point in English... :( I tried to update it few examples, but... 🤷‍♂️

The _Default_ buttons are visually light (greyish) and you can have as many of them as you want (within common sense :D). 
They can (but don't have to) be used in conjunction with a _Primary_ button and less often with _Secondary_ buttons (use your common sense again or ping your designer).
When combined with the _Primary_ or the _Secondary_, the _Default_ variant should be used for any remaining actions.

Whenever used, make sure it has sufficient color contrast against the background it's used on.

### Primary

> The _Primary_ variant has special look: high contrast (against rest of the view/component) and it's filled with accent color. 

Reviewer 2: **_Quick suggestion_**: there's an "a" missing here. 😊 
```suggestion
The _Primary_ variant has a special look: high contrast (against rest of the view/component) and it's filled with accent color. 
```

Ideally there should be only one _Primary_ button per entire component, or sometimes even per entire view.
Primary buttons should be used for the principal action on the page.

### Secondary

> The _Secondary_ variant is visually in between _Primary_ and _Default_ buttons. It's not as loud as _Primary_ button but definitely stands out more than _Default_ button.

Reviewer 2: **_Quick suggestion_**: let's add "the" before the button names here for consistency and readability. 😊 
```suggestion
The _Secondary_ variant is visually in between the _Primary_ and the _Default_ buttons. It's not as loud as the _Primary_ button but definitely stands out more than the _Default_ button.
```

It's especially useful when you want your button to be more distinguishable from its surroundings without making it super loud (like _Primary_). 
A good example would be a component with lots of text. The _Default_ button could be potentially missed there (as it uses similar colors as regular texts).

Whenever you have doubts between using the _Secondary_ vs the _Default_ variants, the rule of thumb should be:

> Use the _Default_ variant by default :D unless it's not distinguishable enough from its surroundings.

When combined with the _Primary_ button, the _Secondary_ variant should be used for any remaining or alternative actions.

## Configuration

### Destructive actions

For actions that could have destructive effects on the user’s data (for example, *delete* or *remove*) we have a special style.
This style can be applied by using `destructive` prop and it will work with two variants: _Primary_ and _Default_ (as it doesn't really make sense to have _Secondary_ variant for destructive buttons).

### Icons

> Icon can be placed in front of a label to both clarify an action but also to make button stand out more.

Reviewer 2: Quick suggestion: let's make "icon" plural here and add the word "the"!
```suggestion
Icons can be placed in front of a label to both clarify an action but also to make the button stand out more.
```

Author: But you can only have one icon in a button?
